### PR TITLE
Add refresh druid datasources endpoint.

### PR DESCRIPTION
**Pull request by bkyryliuk**
Provides the endpoint to refresh the druid datasources for the cases when the druid datasource schema is stored outside of the druid cluster.

Configuration should include:
- `name`: datasource_name
- `dimensions`: list of the dimensions, they become druid columns with the type STRING
- `metrics_spec`: list of `metrics` (dictionary). `metric` consists of 2 attributes: `type` and `name`. `type` can be `count`, etc. `count` type is stored internally as `longSum`
- other fields will be ignored.

Example config:

```
{
  "name": "test_click", 
  "metrics_spec": [{"type": "count", "name": "count"}], 
  "dimensions": ["affiliate_id", "campaign", "first_seen"]
}
```

Reviewers:
- @mistercrunch 
- @kim-pham-airbnb 
- @hongbozeng
- @vera-liu 
- @ascott

### CODE REVIEW

File: tests/core_tests.py [hunk 1]
```diff
@@ -212,6 +212,38 @@ def test_add_slice_redirect_to_druid(self, username='admin'):
...
+                    "partition": "d='{{ ds }}'"
+                }]
+            }
+        }
+        resp = self.client.post(
+            '/caravel/sync_druid/', data=dict(
+                config=json.dumps(cfg), user="admin", cluster="new_druid"))
+
+        druid_ds = db.session.query(DruidDatasource).filter_by(
+            datasource_name="test_click").first()
```
Comment (chatbots-airbnb-2016) [inline tests/core_tests.py]: fit in one line? here and a couple of other places as well.

File: tests/core_tests.py [hunk 2]
```diff
@@ -212,6 +212,38 @@ def test_add_slice_redirect_to_druid(self, username='admin'):
...
+        assert druid_ds
+        column_names = [c.column_name for c in druid_ds.columns]
+        assert set(column_names) == set(
+            ["affiliate_id", "campaign", "first_seen"])
+        metric_names = [m.metric_name for m in druid_ds.metrics]
+        assert set(metric_names) == set(["count"])
+        assert resp.status_code == 201
```
Comment (kim-pham-airbnb) [inline tests/core_tests.py]: what happens if the data is already there? Return 200 instead?

Comment (kim-pham-airbnb) [inline tests/core_tests.py]: Could you please add a test when you update some columns or metrics?

**Comment by kim-pham-airbnb** [comment]
This looks great @bkyryliuk ! 

I'll let @hongbozeng to have the final stamp.

File: tests/core_tests.py (hunk 1 above)
Comment (bkyryliuk) [inline tests/core_tests.py]: pep8 limit is 80 lines.

**Comment by bkyryliuk** [comment]
@kim-pham-airbnb - update the test cases.
Metrics and dimensions won't be overridden or removed as it was in the original implementation.

**Comment by chatbots-airbnb-2016** [comment]
lg tm !

@hongbozeng this basically reimplements our old script https://git.musta.ch/airbnb/data-infra/blob/master/airolap/utils/sync_to_panoramix.py in caravel.

**Comment by kim-pham-airbnb** [comment]
sorry I keep using the wrong github account 🐷

**Comment by kim-pham-airbnb** [comment]
since Hongbo is busy, i'm gonna 👍  this. 🚢  it @bkyryliuk

**Comment by bkyryliuk** [comment]
thanks @kim-pham-airbnb for the review!

**Comment by maver1ck** [comment]
Any documentation for that ?

**Pull request merged**